engine/src/main/java/com/redhat/cloud/notifications/recipients/rbac/RbacRecipientUsersProvider.java

```diff
@@ -77,20 +93,47 @@ public void initCounters() {
                     LOGGER.warnf("RBAC S2S call failed", event.getException().getMessage());
                 })
                 .build();
+
+        itFailuresCounter = meterRegistry.counter("it.failures");
+        itRetryPolicy = RetryPolicy.builder()
+                .onRetry(event -> itFailuresCounter.increment())
+                .handle(IOException.class, ConnectTimeoutException.class)
+                .withBackoff(initialBackOffIt, maxBackOffIt)
+                .withMaxAttempts(maxRetryAttemptsIt)
+                .onRetriesExceeded(event -> {
+                    // All retry attempts failed, let's log a warning about the failure.
+                    LOGGER.warnf("IT User Service call failed", event.getException().getMessage());
+                })
+                .build();
     }
 
     @CacheResult(cacheName = "rbac-recipient-users-provider-get-users")
     public List<User> getUsers(String accountId, boolean adminsOnly) {
         Timer.Sample getUsersTotalTimer = Timer.start(meterRegistry);
+
         List<User> users;
         if (retrieveUsersFromIt) {
-            final List<ITUserResponse> itUserResponses = itUserService.getUsers(accountId, adminsOnly);
-            users = transformToUser(accountId, itUserResponses, adminsOnly);
+            List<ITUserResponse> usersPaging;
+            List<ITUserResponse> usersTotal = new LinkedList<>();
+
+            int firstResult = 0;
+
+            do {
+                int finalFirstResult = firstResult;
+                usersPaging = retryOnItError(() -> itUserService.getUsers(accountId, adminsOnly, finalFirstResult, maxResultsPerPage));
+                usersTotal.addAll(usersPaging);
+
+                firstResult += maxResultsPerPage;
+            } while (usersPaging.size() == maxResultsPerPage);
+
+            getUsersTotalTimer.stop(meterRegistry.timer("rbac.get-users.total", "accountId", accountId, "users", String.valueOf(usersTotal.size())));
```

Comment:
I just noticed this line is a duplicate of the same instruction at the end of this method:
```suggestion
```
The `transformToUser` execution time is close to zero so it doesn't matter if it is counted by the timer and we will refactor things anyway when the RBAC implementation will be removed.